azure-pipelines.release-fluentui.yml

```
@@ -1,74 +1,220 @@
-  - script: |
-      git config user.name "Fluent UI Build"
-      git config user.email "fluentui-internal@service.microsoft.com"
-      git remote set-url origin https://$(githubUser):$(githubPAT)@github.com/microsoft/fluentui.git
-    displayName: Authenticate git for pushes
-  - task: Bash@3
-    inputs:
-      filePath: yarn-ci.sh
-    displayName: yarn
-  - script: |
-      PRODUCTION=true yarn build
-    displayName: yarn build
-  - script: |
-      NODE_ENV=production yarn build:fluentui:docs
-    displayName: yarn build:fluentui:docs
-  - task: PublishBuildArtifacts@1
-    displayName: 'Publish docsite as pipeline artifact'
-    inputs:
-      PathtoPublish: 'packages/fluentui/docs/dist'
-      ArtifactName: 'docsite_v$(docsiteVersion)'
-
-  - task: AzureUpload@2
-    displayName: Upload docs site
-    inputs:
-      SourcePath: 'packages/fluentui/docs/dist'
-      azureSubscription: 'UI Fabric (private)'
-      storage: fluentsite
-      ContainerName: '$web'
-      BlobPrefix: '$(deployBasePath)'
-      # uncomment to deploy to prod
-      # storage: fluentsite
-      # BlobPrefix: ''
-# No NPM release in CI for now
-#  - script: |
-#      yarn test
-#    displayName: yarn test
-#
-#  - script: |
-#      yarn lint
-#    displayName: yarn lint
-#
-#  - script: |
-#      yarn release:fluentui:$(bumpType)
-#    displayName: yarn release:fluentui:$(bumpType)
+    value: true
+  - name: nightlyRelease
+    value: true
+  # below are variables configurable at queue time, so they cannot appear in yml file
+  # - name: publishDocsiteOnly # set it to true to only run Job_2
+  #   value: false
+  # - name: publishOfficial # set it to true to publish package to npm, and publish docsite for official releases
+  #   value: false
+  # - name: version # version to publish to npm; only make sense when publishOfficial=true
+  #   value: canary # value can only be canary/patch/minor
+
+jobs:
+  - job: Job_1
+    displayName: Build and Release Fluent Packages
+    # skip this job if publishDocsiteOnly is true
+    condition: and(succeeded(), eq(variables.publishDocsiteOnly, false))
+    pool:
+      name: Self Host Ubuntu
+
+    steps:
+      - checkout: self
+
+      - task: CmdLine@2
+        displayName: Authenticate Git credentials
+        inputs:
+          script: >-
+            git config user.name "Fluent UI Build"
+
+            git config user.email "fluentui-internal@service.microsoft.com"
+
+            git remote set-url origin https://$(githubUser):$(githubPAT)@github.com/microsoft/fluentui.git
+
+      - task: CmdLine@2
+        displayName: Checkout branch
+        inputs:
+          script: >-
+            git checkout $(Build.SourceBranchName)
+
+            git pull
+
+      - task: NodeTool@0
+        displayName: Install Node.js
+        inputs:
+          versionSpec: 12.x
+
+      - task: YarnInstaller@0
+        displayName: Use Yarn 1.19.x
+        inputs:
+          versionSpec: 1.19.x
+          checkLatest: true
```

```suggestion
      - template: .devops/templates/tools.yml
```

We have a template for this